hypothesis-python/src/hypothesis/internal/constants_ast.py

```diff
@@ -96,22 +93,23 @@ def visit_Constant(self, node):
         self.generic_visit(node)
 
 
-@lru_cache(1024)
-def constants_from_ast(tree: AST) -> AbstractSet[ConstantT]:
-    visitor = ConstantVisitor()
-    visitor.visit(tree)
-    return visitor.constants
-
-
-@lru_cache(1024)
-def _module_ast(module: ModuleType) -> Optional[AST]:
+@lru_cache(4096)
+def constants_from_module(module: ModuleType) -> AbstractSet[ConstantT]:
     try:
         source = inspect.getsource(module)
         tree = ast.parse(source)
     except Exception:
-        return None
+        return set()
 
-    return tree
+    visitor = ConstantVisitor()
+    try:
+        visitor.visit(tree)
+    except RecursionError:  # pragma: no cover
+        # NodeVisitor has a recursive structure, and so deeply nested expressions
+        # can hit the recursion limit. See the many related issues on e.g. libcst:
+        # https://github.com/Instagram/LibCST/issues?q=recursion
+        pass
```

Comment:
It's not just `RecursionError` either; e.g.
```python-traceback
>>> _ = ast.parse("[1, " * (x:=200) + "]" * x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python3.12/ast.py", line 52, in parse
    return compile(source, filename, mode, flags,
MemoryError: Parser stack overflowed - Python source too complex to parse
```
so I think it'd just put this whole function inside the main `try:`/`except Exception:` block.